test/gpu/native/interop/gpuLibrary/udf2.chpl

```diff
@@ -0,0 +1,17 @@
+module Foo {
+
+  extern proc chpl_task_getRequestedSubloc(): int(32);
+  extern proc printf(s...);
+
+  export proc add_int32(ref result: [] int(32), const ref a: [] int(32),
+                        const ref b: [] int(32)) {
+    printf("before on\n");
+    on here.gpus[0] { // `sync begin on` is needed
+      printf("subloc : %d\n", chpl_task_getRequestedSubloc()); // prints -2 without `sync begin on`
+      @assertOnGpu
```

Comment:
how does this not fail without `sync begin on`?

Reply:
It doesn't with CHPL_GPU=cpu :smile:. I realized the problem when testing `CHPL_GPU=nvidia`

Reply:
> t doesn't with CHPL_GPU=cpu 

... "and the other environment variable that suppresses the error-turned-into-warning in the CPU-as-device mode" is I am hoping the rest of this sentence :)